File: exercism/python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if number < 1:
        raise ValueError('The number must be greater than 0')

    steps = 0
    while number > 1:
        steps += 1

        if number%2 == 0:
            number //= 2
        else:
            number = 3*number + 1

    return steps

File: exercism/go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "fmt"

// CollatzConjecture returns the number of steps needed to perform the collatz
// conjecture on n
func CollatzConjecture(n int) (int, error) {
	var steps int

	if n < 1 {
		return -1, fmt.Errorf("n must be larger than 0")
	}

	for n > 1 {
		if n&1 == 0 {
			n >>= 1
		} else {
			n = 3*n + 1
		}

		steps++
	}

	return steps, nil
}
